### PR TITLE
Refactor the ImportDate ActiveModel class & update our session management logic

### PR DESCRIPTION
**Pull request by tavy88**
### Jira link

HOTT-441

### What?

I have added/removed/altered:

- [x] Refactor the ImportDate ActiveModel class
- [x] Remove all the keys from the session that follow the current question

### Why?

We needed to be able to prepopulate the date_select field
with the value from the session.

This is needed for when a user comes back to this question.
In addition, we needed an invalid past date to be persisted
on the form so that users can see what they've done wrong.

Also updating the error message.

### CODE REVIEW

File: app/models/wizard/steps/base.rb [hunk 1]
```diff
@@ -9,12 +9,26 @@ class Base
       def initialize(session, attributes = {})
         @session = session
 
+        clean_session if attributes.empty?
```
Comment (tavy88) [inline app/models/wizard/steps/base.rb:12]: Ensure we only clean the session while on show action. attributes will be non-empty only on #create actions.

File: app/models/wizard/steps/base.rb [hunk 2]
```diff
@@ -9,12 +9,26 @@ class Base
...
+      private
+
+      def clean_session
+        return if keys_to_remove.empty?
+
+        session.delete(*keys_to_remove)
+      end
+
+      def keys_to_remove
+        session.keys.map(&:to_i).uniq.select { |key| key > self.class::STEP_ID }
```
Comment (willfish) [inline app/models/wizard/steps/base.rb:30]: Exactly :)

File: app/models/wizard/steps/import_date.rb [hunk 3]
```diff
@@ -1,38 +1,68 @@
...
+
       attribute 'import_date(3i)', :string
       attribute 'import_date(2i)', :string
       attribute 'import_date(1i)', :string
 
       validate :import_date_in_future
 
-      def import_date
-        session[:import_date]
+      STEP_ID = 1
```
Comment (willfish) [inline app/models/wizard/steps/import_date.rb:14]: Do we want this to be a string?

Comment (willfish) [reply app/models/wizard/steps/import_date.rb:14]: I don't think it'll get serialised as an integer

Comment (willfish) [reply app/models/wizard/steps/import_date.rb:14]: So we'll be mixing types.

File: app/views/wizard/steps/import_dates/show.html.erb [hunk 4]
```diff
@@ -3,7 +3,7 @@
 <main class="govuk-main-wrapper govuk-main-wrapper--auto-spacing" id="main-content" role="main">
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-three-quarters">
-      <%= form_for @import_date, builder: GOVUKDesignSystemFormBuilder::FormBuilder, url: import_date_path do |f| %>
+      <%= form_for @import_date, builder: GOVUKDesignSystemFormBuilder::FormBuilder, url: import_date_path, html: { novalidate: true } do |f| %>
```
Comment (willfish) [inline app/views/wizard/steps/import_dates/show.html.erb:6]: I'm assuming we're skipping validations that are done automatically here? Not sure I understand this change

File: .gitignore [hunk 5]
```diff
@@ -11,6 +11,9 @@
 /log/*
 /tmp/*
 
+# Ignore simplecov files
+/coverage/*
```
Comment (willfish) [inline .gitignore:15]: Would be good at some point to plumb this into the duty calc

File: app/models/wizard/steps/import_date.rb (hunk 3 above)
Comment (tavy88) [reply app/models/wizard/steps/import_date.rb:14]: Uhm sure yeah.

File: app/views/wizard/steps/import_dates/show.html.erb (hunk 4 above)
Comment (tavy88) [reply app/views/wizard/steps/import_dates/show.html.erb:6]: It's something that GDS wants.

File: .gitignore (hunk 5 above)
Comment (tavy88) [reply .gitignore:15]: Do you mean CircleCi wise? Think Miguel is on it already.

**Review by willfish — APPROVED** [review]

**Pull request merged**